docs/src/main/sphinx/security/oauth2.rst

```diff
@@ -0,0 +1,126 @@
+========================
+OAuth 2.0 authentication
+========================
+
+Trino can be configured to enable OAuth 2.0 authentication over HTTPS for the
+Web UI and the JDBC driver. Trino use `Authorization Code
+<https://tools.ietf.org/html/rfc6749#section-1.3.1>`_ flow which exchanges an
+Authorization Code for a token. The flow includes, at a high level, following
+steps:
+
+#. the Trino coordinator redirects a user agent to the Authorization Server
+#. the user authenticates in the Authorization Server and approves the Trino's permissions request
+#. the user agent is redirected back to the Trino coordinator with an authorization code
+#. the Trino coordinator exchanges the authorization code for a token
+
+To enable OAuth 2.0 authentication for Trino, configuration changes are made on
+the Trino coordinator. No changes are required to the worker configuration;
+only the communication from the clients to the coordinator is authenticated.
+However, if you want to secure the communication between
+Trino nodes with SSL/TLS configure :doc:`/security/internal-communication`.
+
+Trino server configuration
+--------------------------
+
+Trino coordinator node configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Access to the Trino coordinator should be through HTTPS. You can do that
```

Comment:
dont explain here how to set up TLS.

Just say something like

Using the OAuth2 authentication requires your coordinator to be secured with TLS.

We are working on a new TLS docs that will hopefully also make it in for the 353 release and we can add a link there to it later.

cc @Ordinant @electrum @dain